src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -367,4 +370,78 @@ public boolean contains(String key) {
 
         return mPrefs.contains(key);
     }
+    
+    private TokenCacheItem getFamilyRefreshTokenItemForUser(String userId) {       
+        
+        Iterator<TokenCacheItem> results = this.getTokensForUser(userId).iterator();
+        TokenCacheItem tokenItem = new TokenCacheItem();   
+        while (results.hasNext()) {
+            final TokenCacheItem tokenCacheItem = results.next();
+            if(tokenCacheItem.getUserInfo() != null
+                    && !StringExtensions.IsNullOrBlank(tokenCacheItem.getFamilyClientId())
+                    && tokenCacheItem.getFamilyClientId().equalsIgnoreCase("1")){
+                tokenItem = tokenCacheItem;
+                //return the cacheItem which contains the FRT for this userId
+                //there should be at most only one FRT record for each userId
+                return tokenItem;
+            }
+        }
+        //If the cache does not contain the FRT for this user return null
+        Logger.i(TAG, "Cannot find the family refresh token cache item for this user in cache", "");
+        return null;
+    }
+    
+    protected String serialize(String uniqueUserId) throws UsageAuthenticationException {
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if((new BrokerProxy(mContext)).canSwitchToBroker()){
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
```

Comment:
Hi @imakhalova , thank you for your reviews. Do you mean we should not throw any exception if the developer calls serialize() while using the broker? This is different from the requirement on the aadwiki. @weijjia